util/reflect_test.go

```diff
@@ -86,6 +86,163 @@ func toStringPtr(s string) *string { return &s }
 func toInt8Ptr(i int8) *int8       { return &i }
 func toInt32Ptr(i int32) *int32    { return &i }
 
+func TestIsValueNil(t *testing.T) {
+	if !IsValueNil(nil) {
+		t.Error("got IsValueNil(nil) false, want true")
+	}
+	if !IsValueNil((*int)(nil)) {
+		t.Error("got IsValueNil(ptr) false, want true")
+	}
+	if !IsValueNil((map[int]int)(nil)) {
+		t.Error("got IsValueNil(map) false, want true")
+	}
+	if !IsValueNil(([]int)(nil)) {
+		t.Error("got IsValueNil(slice) false, want true")
+	}
+	if !IsValueNil((interface{})(nil)) {
+		t.Error("got IsValueNil(interface) false, want true")
+	}
+
+	if IsValueNil(toInt8Ptr(42)) {
+		t.Error("got IsValueNil(ptr) true, want false")
+	}
+	if IsValueNil(map[int]int{42: 42}) {
+		t.Error("got IsValueNil(map) true, want false")
+	}
+	if IsValueNil([]int{1, 2, 3}) {
+		t.Error("got IsValueNil(slice) true, want false")
+	}
+	if IsValueNil((interface{})(42)) {
+		t.Error("got IsValueNil(interface) true, want false")
+	}
+}
+
+// TODO(mostrowski): check interface types.
+
+func TestIsValueFuncs(t *testing.T) {
```

Comment:
I added a bunch of different tests for the `IsValue...` and `IsType...` in https://github.com/openconfig/ygot/blob/master/util/reflect_test.go#L557 -- can you PTAL since it seems like we have some duplication of testing here?

Reply:
I've merged the two test sets - I think this format exercises a few more type/function combos. Let me know if something got lost in the process.

Reply:
I'm happy with this format -- just figured we didn't necessarily want two sets of tests that covered exactly the same thing.